test/linalg/symmetric.jl

```diff
@@ -42,206 +41,229 @@ end
+                end
+                
+                @testset "det" begin
+                    @test det(asym) ≈ det(Hermitian(asym, :U))
+                    @test det(asym) ≈ det(Hermitian(asym, :L))
+                    if eltya <: Real
+                        @test det(asym) ≈ det(Symmetric(asym, :U))
+                        @test det(asym) ≈ det(Symmetric(asym, :L))
+                    end
+                    @test det(a + a.') ≈ det(Symmetric(a + a.', :U))
+                    @test det(a + a.') ≈ det(Symmetric(a + a.', :L))
+                end
+
+                @testset "rank" begin
+                    let A = a[:,1:5]*a[:,1:5]'
+                        # Make sure A is Hermitian even in the present of rounding error
+                        # xianyi/OpenBLAS#729
+                        A = (A' + A) / 2
+                        @test rank(A) == rank(Hermitian(A))
+                    end
+                end
+
+                @testset "pow" begin
+                    if eltya <: Real && eltya != Int
+                        @test (a*a')^2 ≈ Array(Symmetric(a*a')^2)
+                        @test (a*a')^-2 ≈ Array(Symmetric(a*a')^-2)
+                        @test (a*a')^2.0 ≈ real(Array(Symmetric(a*a')^2.0)) rtol=10*n^2*eps(eltya)
+                        @test (a*a')^-2.0 ≈ real(Array(Symmetric(a*a')^-2.0)) rtol=10*n^2*eps(eltya)
+                    elseif eltya <: Complex
+                        @test (a*a')^2 ≈ Array(Hermitian(a*a')^2)
+                        @test (a*a')^-2 ≈ Array(Hermitian(a*a')^-2)
+                        @test (a*a')^2.0 ≈ Array(Hermitian(a*a')^2.0) rtol=10*n^2*eps(real(eltya))
+                        @test (a*a')^-2.0 ≈ Array(Hermitian(a*a')^-2.0) rtol=10*n^2*eps(real(eltya))
+                    end
+                end
+            end
 
-        # conversion
-        @test Symmetric(asym) == convert(Symmetric,Symmetric(asym))
-        if eltya <: Real && eltya != Int
-            typs = [Float16,Float32,Float64]
-            for typ in typs
-                @test Symmetric(convert(Matrix{typ},asym)) == convert(Symmetric{typ,Matrix{typ}},Symmetric(asym))
+            @testset "isposdef[!]" begin
+                @test isposdef(Symmetric(asym)) == isposdef(full(Symmetric(asym)))
+                @test isposdef(Hermitian(asym)) == isposdef(full(Hermitian(asym)))
+                if eltya != Int
+                    @test isposdef!(Symmetric(copy(asym))) == isposdef(full(Symmetric(asym)))
+                    @test isposdef!(Hermitian(copy(asym))) == isposdef(full(Hermitian(asym)))
+                end
             end
         end
-        if eltya <: Complex && eltya != Int
-            typs = [Complex64,Complex128]
-            for typ in typs
-                @test Hermitian(convert(Matrix{typ},asym)) == convert(Hermitian{typ,Matrix{typ}},Hermitian(asym))
+        @testset "linalg binary ops" begin
+            @testset "mat * vec" begin
+                if eltya <: Complex
```

Comment:
We can test `Hermitian` and `Symmetric` for both real and complex cases here, perhaps with a loop over `Hermitian` and `Symmetric`?